### PR TITLE
Adding opt-out for telemetry r=ferjm

### LINKED ISSUE #221: Add to settings the avility to opt-out from the telemetry system

### PR DESCRIPTION
**Pull request by arcturus**
Closes #221 

@ferjm would you mind taking a look?

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -20,6 +20,12 @@ module.exports = {
       value: false, // default
       title: 'Remove old items',
     },
+
+    "enableTelemetry": {
+      value: true, // default
+      title: 'Telemetry',
+      msg: 'Shares performance and usage data with Mozilla to help us to make Project Magnet better'
```
Comment (ferjm) [inline config.js]: uber-nits:
- s/Shares/Share
- "...to help us improve Project Magnet"

File: lib/app.js [hunk 2]
```diff
@@ -70,8 +69,10 @@ class App extends React.Component {
       getItems: () => this.props.items
     });
 
-    this.setupNotificationListeners();
-    this.inflateUserFlags();
+    this.inflateUserFlags().then(() => {
```
Comment (ferjm) [inline lib/app.js:74]: Maybe add a comment here explaining that we need to fetch user prefs before setting up the tracker.

File: lib/utils/tracker.js [hunk 3]
```diff
@@ -7,8 +7,12 @@
 const ga = require('react-native-google-analytics-bridge');
 const { analyticsTrackerId } = require('../../config');
```
Comment (ferjm) [inline lib/utils/tracker.js]: You are getting the entire `config` object on line 13. I guess we can save one `require` call.

File: lib/utils/tracker.js [hunk 4]
```diff
@@ -18,20 +22,36 @@ class Tracker {
...
 
+  _isEnabled() {
+    const value = store.getState() ?
+      store.getState().userFlags.enableTelemetry.value :
+      TELEMETRY_DEFAULT_ENABLED;
+    return value;
+  }
+
   view(name) {
     debug('view', name);
```
Comment (ferjm) [inline lib/utils/tracker.js:33]: Do we want to log debug messages even if `!this._isEnabled`?

File: lib/views/settings-scene.js [hunk 5]
```diff
@@ -88,15 +88,20 @@ class SettingsScene extends React.Component {
 
   renderUserFlags(flags) {
     return Object.keys(flags).map(key => {
-      var { value, title } = flags[key];
+      var { value, title, msg } = flags[key];
       debug('render user flag', key, value);
+      var explanation = undefined;
+      if (msg) {
+        explanation = <Text style={styles.listSubItemText}>{msg}</Text>;
+      }
```
Comment (ferjm) [inline lib/views/settings-scene.js]: Maybe `var explanation = msg ? <Text style={styles.listSubItemText}>{msg}</Text> : undefined;`

**Comment by ferjm** [comment]
LGTM!

File: lib/app.js (hunk 2 above)
Comment (arcturus) [reply lib/app.js:74]: Definitely.

File: lib/utils/tracker.js (hunk 3 above)
Comment (arcturus) [reply lib/utils/tracker.js]: Yup! I knew i was missing the config previously. Good catch.

File: lib/utils/tracker.js (hunk 4 above)
Comment (arcturus) [reply lib/utils/tracker.js:33]: Yes, the debug message won't do anything on a production build, and want to leave it there even if the telemetry is off

File: config.js (hunk 1 above)
Comment (arcturus) [reply config.js]: Sounds good to me, just got the wording from firefox telemetry :)

**Pull request merged**